beangulp/petl_utils.py

```diff
@@ -47,32 +47,33 @@ def table_to_directives(
             metas.append((column, match.group(1)))
 
     # Create transactions.
-    entries = []
+    entries: data.Entries = []
     filename = filename or f"<{__file__}>"
     for index, rec in enumerate(table.records()):
         meta = data.new_metadata(filename, index)
         units = amount.Amount(rec.amount, currency)
-        tags, links = set(), set()
+        tags: Set[str] = set()
+        links: Set[str] = set()
+        link = getattr(rec, "link", None)
+        if link:
+            links.add(link)
+        tag = getattr(rec, "tag", None)
+        if tag:
+            tags.add(tag)
```

Comment:
I'm not entirely convinced that the type for `links` and `tags` should be `frozenset` and not simply `collections.abc.Set`. Anyhow,  This seems a very convoluted way to crete a `frozenset` of one element.

Reply:
I stuck to the original code but I'm happy to refactor this as well